SDK v2 migration guide.md

```diff
@@ -0,0 +1,399 @@
+| ModuleClient#subscribeToTwinDesiredProperties(Map<Property, Pair<TwinPropertyCallBack, Object>>(); | ModuleClient#subscribeToDesiredPropertiesAsync(DesiredPropertiesSubscriptionCallback, Object, DesiredPropertiesCallback, Object);  |
+| ModuleClient#sendReportedProperties(Set<Property>); | ModuleClient#updateReportedPropertiesAsync(TwinCollection, ReportedPropertiesCallback, Object);  |
+| ModuleClient#sendReportedProperties(Set<Property>, int); | ModuleClient#updateReportedPropertiesAsync(TwinCollection, ReportedPropertiesCallback, Object);  |
+| ModuleClient#sendReportedProperties(ReportedPropertiesParameters); | ModuleClient#updateReportedPropertiesAsync(TwinCollection, ReportedPropertiesUpdateCorrelatingMessageCallback, Object);  |
+| ModuleClient#sendReportedProperties(Set<Property>, Integer, CorrelatingMessageCallback, Object, IotHubEventCallback, Object); | ModuleClient#updateReportedPropertiesAsync(TwinCollection, ReportedPropertiesUpdateCorrelatingMessageCallback, Object);  |
+| ModuleClient#setOption(String, Object); | no equivalent method***  |
+| ModuleClient#setProxySettings(ProxySettings); | no equivalent method****  |
+
+*** The options that were previously set in this method are now set at ModuleClient constructor time in the optional ClientOptions parameter.
+
+**** Proxy settings are now set at ModuleClient constructor time in the optional ClientOptions parameter,
+
+#### MultiplexingClient
+
+No API surface changes have been made to the MultiplexingClient class
+
+#### TransportClient
+
+This client has been removed in v2. It is replaced by the MultiplexingClient. See this repo's multiplexing client sample for more information.
+
+#### Other notable breaking changes
+
+- Trust certificates are read from the physical device's trusted root certification authorities certificate store rather than from source.
+  - Users are expected to install the required public certificates into this certificate store if they are not present already.
+  - See [this document](./upcoming_certificate_changes_readme.md) for additional context on which certificates need to be installed.
+- DeviceClient and ModuleClient constructors that take public certificates and private keys as strings have been removed.
+  - Users must provide an instance of SSLContext that has their public certificates and private keys loaded into it instead.
+  - See [this sample](./device/iot-device-samples/send-event-x509) for the recommended way to create this SSLContext and how to construct your DeviceClient and/or ModuleClient with it.
+- deviceClient.uploadToBlobAsync() has been removed.
+  - Users can still use deviceClient.getFileUploadSasUri() to get a SAS URI that can be used with the Azure Storage SDK to upload the file.
+  - See [this sample](./device/iot-device-samples/file-upload-sample) for the recommended way to upload files.
+- The Bouncycastle, JNR Unixsocket, and Azure Storage SDK dependencies have been removed.
```

Comment:
Q - do we want to explain what these libraries did and what we are doing now, or atleast call out that we are now implementing <xyz> natively, so essentially there is no loss of functionality?

Reply:
Sure, good idea